katz/src/main/kotlin/katz/data/Reader.kt

```diff
@@ -7,20 +7,20 @@ infix fun <A, B, C> ((A) -> B).andThen(g: (B) -> C): (A) -> C =
         { a: A -> g(this(a)) }
 
 fun <D, A> ((D) -> A).reader(): ReaderT<Id.F, D, A> =
-        Reader(this)
+        Reader(this, Id)
 
 fun <D, A> ReaderT<Id.F, D, A>.runId(d: D): A =
         this.run(d).value()
 
 object Reader {
 
-    operator fun <D, A> invoke(run: (D) -> A): ReaderT<Id.F, D, A> =
-            Kleisli(run.andThen { Id(it) })
+    inline operator fun <D, A> invoke(noinline run: (D) -> A, MF: Monad<Id.F> = monad<Id.F>()): ReaderT<Id.F, D, A> =
```

Comment:
There is only one possible instance of this monad, so I'd be explicit about it instead.

MF: Monad<Id.F> = Id